### DISCUSSION TITLE
Add --sync to global flags

### DISCUSSION THREAD

**Discussion by avoidik** [astral-sh/rye#584] [Ideas]
Hello from a curious person who sees Rye for the first time!

I was exploring a rye a bit and came to a conclusion that almost all operations with the cli are revolving around specific pattern, like:
- `rye [cmd]`
- `rye sync`

wouldn't make sense to introduce a global flag `--sync` so that we can optimize it just to one step, like:

- `rye [cmd] --sync`

meanwhile `rye sync` stays as is

what do you think?

**Comment by mitsuhiko** [comment]
My desire is to eventually make sync in most cases automatic. This will require a dramatically faster installation process though.
